torch/distributed/fsdp/_optim_utils.py

```diff
@@ -475,12 +458,21 @@ def _flatten_optim_state_dict(
             if flat_state:
                 flat_osd_state[key] = flat_state
             elif optim is not None:  # NamedOptimizer or KeyedOptimizer case.
-                assert len(fqns) == 1
-                local_wrapped_fqn = local_state_clean_fqns.get(fqn, "")
-                if local_wrapped_fqn:
-                    flat_osd_state[key] = copy.deepcopy(
-                        local_state_dict[local_wrapped_fqn]
+                assert len(fqns) == 1 and use_orig_params
```

Comment:
nit: Should we add an assert message here just in case? For example, it may be helpful to at least include `"{len(fqns)} {fqns}` (so we know whether it was `len(fqns) !=` or `not use_orig_params` and have some sense of the FQNs).